frontend/src/services/GraphDataSource.ts

```diff
@@ -197,6 +199,12 @@ export class GraphDataSource {
       appenders += ',idleNode';
     }
 
+    if (!fetchParams.node && !fetchParams.showWaypoint) {
+      // note we only use the idleNode appender if this is NOT a drilled-in node graph and
+      // the user specifically requests to see idle nodes.
+      appenders += ',hideWaypoint';
```

Comment:
Let's talk about this some more.  I'm not sure that we need an appender for this if we assume that we will have dedicated graph generation for ambient namespaces.  What I mean is that I *think* in the near future we may want to have code that is much more specific to ambient, to be more correct and to be more efficient.

Or, perhaps we will have a more general appender called "ambient" that takes various "appender request parameters" (like, for example the way the response_time appender gets the desired quantile as a param).

Reply:
I'm sure we will have more things we may want to for Ambient and so, for now at least, let's make this a little more generic and just introduce an "ambient" appender that can perform the optional waypoint logic, and possibly other things as well.  I'll re-review and add specific comments...